File: my-folder/0003-longest-substring-without-repeating-characters/solution.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l, r = 0, 0
        maxLen = 0
        existing = set()
        while r < len(s):
            if s[r] not in existing:
                existing.add(s[r])
                maxLen = max(maxLen, r - l + 1)
                r += 1
            else:
                existing.remove(s[l])
                l += 1
        return maxLen

File: my-folder/0003-longest-substring-without-repeating-characters/solution.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<int> set = new HashSet<int>();
        int l = 0, r = 0;
        int res = 0;
        // set.Add(s[0]);
        while (r < s.Length) {
            if (set.Contains(s[r])) {
                set.Remove(s[l]);
                l++;
                // while(set.Contains(s[r])) {
                // }
            } else {
                set.Add(s[r]);
                res = Math.Max(res, r-l+1);
                r ++;
            }
        }
        return res;
    }
}
